crates/bevy_ecs/src/relationship/mod.rs

```diff
@@ -128,6 +128,33 @@ pub trait Relationship: Component + Sized {
             world.commands().entity(entity).remove::<Self>();
             return;
         }
+        // For one-to-one relationships, remove existing relationship before adding new one
+        let mut current_source_to_remove: Option<Entity> = None;
+        if core::any::TypeId::of::<<Self::RelationshipTarget as RelationshipTarget>::Collection>()
+            == core::any::TypeId::of::<Entity>()
+        {
+            if let Ok(target_entity_ref) = world.get_entity(target_entity) {
+                if let Some(relationship_target) =
+                    target_entity_ref.get::<Self::RelationshipTarget>()
+                {
+                    // SAFETY: We just checked that the collection type is Entity
+                    let entity_collection = unsafe {
+                        &*core::ptr::from_ref::<
+                            <Self::RelationshipTarget as RelationshipTarget>::Collection,
+                        >(relationship_target.collection())
+                        .cast::<Entity>()
+                    };
+                    if *entity_collection != Entity::PLACEHOLDER {
+                        current_source_to_remove = Some(*entity_collection);
+                    }
+                }
+            }
+        }
+
+        if let Some(current_source) = current_source_to_remove {
+            world.commands().entity(current_source).try_remove::<Self>();
```

Comment:
Does this have a race condition if we do a batch insert, similar to the one #19519 fixed?  If we insert multiple one-to-one relationship sources without flushing, this will remove any *previous* source from the target, but I think later sources in the batch will overwrite earlier ones without removing.  

This might be easier to fix if we merge #19932 first and do this whole check inside the command.